src/client/dfs/dfs.c

```diff
@@ -2158,9 +2206,9 @@ dfs_mount(daos_handle_t poh, daos_handle_t coh, int flags, dfs_t **_dfs)
 	int                        amode;
 	int                        rc;
 	int                        i;
-	int                        num_props = 4;
-	uint32_t props[] = {DAOS_PROP_CO_LAYOUT_TYPE, DAOS_PROP_CO_ROOTS, DAOS_PROP_CO_OWNER,
-			    DAOS_PROP_CO_OWNER_GROUP};
+#define NUM_PROPS 3
+	uint32_t props[NUM_PROPS] = {DAOS_PROP_CO_LAYOUT_TYPE, DAOS_PROP_CO_ROOTS,
```

Comment:
Can NUM_PROPS not just be a local int? Can compilers not figure that out?

Reply:
Agreed.  The changes to this file are a cherry-pick from https://github.com/daos-stack/daos/pull/10556 which needs to land first.